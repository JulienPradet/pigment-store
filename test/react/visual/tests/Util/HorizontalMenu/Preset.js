import React from 'react'
import PigmentStore from '../../../../../../dist/index'
import PresetList from '../../../../../../src/core/display/App/util/View/HorizontalMenu/Preset'

const aPresetListConfig = {
  presets: [{
    label: 'S',
    preset: {
      size: {
        width: '360',
        height: 'auto'
      },
      zoom: '100'
    }
  }, {
    label: 'M',
    preset: {
      size: {
        width: '960',
        height: 'auto'
      },
      zoom: '100'
    }
  }, {
    label: 'L',
    preset: {
      size: {
        width: '1920',
        height: 'auto'
      },
      zoom: '50'
    }
  }],
  onSelect: function onSelect (newPreset) {
    console.log(newPreset)
  }
}

export default PigmentStore.React.describe('PresetList', PresetList)
  .featureJsx(
    'Default',
    <PresetList {...aPresetListConfig} />
  )
  .featureJsx(
    'With selected preset',
    <PresetList {...aPresetListConfig} selected={aPresetListConfig.presets[1].preset} />
  )
