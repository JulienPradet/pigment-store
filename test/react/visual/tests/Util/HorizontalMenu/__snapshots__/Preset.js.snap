exports[`PresetList Default 1`] = `
<div
  className="presetList">
  <div
    className="preset">
    <button
      onClick={[Function]}>
      S
    </button>
  </div>
  <div
    className="preset">
    <button
      onClick={[Function]}>
      M
    </button>
  </div>
  <div
    className="preset">
    <button
      onClick={[Function]}>
      L
    </button>
  </div>
</div>
`;

exports[`PresetList With selected preset 1`] = `
<div
  className="presetList">
  <div
    className="preset">
    <button
      onClick={[Function]}>
      S
    </button>
  </div>
  <div
    className="preset presetActive">
    <button
      onClick={[Function]}>
      M
    </button>
  </div>
  <div
    className="preset">
    <button
      onClick={[Function]}>
      L
    </button>
  </div>
</div>
`;
